DuckDuckGo/MacPromoViewController.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  MacPromoViewController.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import SwiftUI
+import DesignResourcesKit
+import DuckUI
+import LinkPresentation
+
+class MacPromoViewController: UIHostingController<MacPromoView> {
+
+    let experiment = MacPromoExperiment()
+
+    convenience init() {
+        self.init(rootView: MacPromoView())
+        rootView.model.controller = self
+        modalPresentationStyle = .pageSheet
+    }
+
+    override var supportedInterfaceOrientations: UIInterfaceOrientationMask {
+        isPad ? .all : .portrait
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+        experiment.sheetWasShown()
+    }
+
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        experiment.sheetWasDismissed()
+    }
+
+    class ViewModel: ObservableObject {
+
+        weak var controller: MacPromoViewController?
+
+        func shareLink() {
+            guard let message = controller?.experiment.message else { return }
```

Comment:
Hardcoding the message works nicely, but when I set my cohort to `sheet` and use the [debug endpoint](https://raw.githubusercontent.com/duckduckgo/remote-messaging-config/brindy/mac-promo-exp2/live/ios-config/ios-config.json), tapping the `Send Link` button doesn't do anything - it's exiting out here as `controller?.experiment.message` is nil

Could this be because the message is marked as dismissed when the sheet is shown? It's fine on the message; the issue is only with the sheet

Reply:
Yep, good spot! Have fixed that by keeping a reference to the model in the view controller.

Also removed the guard in the share link notification observer as that would have prevented the pixel firing after the sheet had closed. 

And once all that worked there was a wee bug here anyway showing the sheet (using the wrong controller's view as the anchor) so fixed that too.